src/dsp_tools/commands/xml_validate/reformat_input.py

```diff
@@ -0,0 +1,213 @@
+from lxml import etree
+
+from dsp_tools.commands.xml_validate.models.data_deserialised import BooleanValueData
+from dsp_tools.commands.xml_validate.models.data_deserialised import ColorValueData
+from dsp_tools.commands.xml_validate.models.data_deserialised import DataDeserialised
+from dsp_tools.commands.xml_validate.models.data_deserialised import DateValueData
+from dsp_tools.commands.xml_validate.models.data_deserialised import DecimalValueData
+from dsp_tools.commands.xml_validate.models.data_deserialised import GeomValueData
+from dsp_tools.commands.xml_validate.models.data_deserialised import GeonameValueData
+from dsp_tools.commands.xml_validate.models.data_deserialised import IntValueData
+from dsp_tools.commands.xml_validate.models.data_deserialised import LinkValueData
+from dsp_tools.commands.xml_validate.models.data_deserialised import ListValueData
+from dsp_tools.commands.xml_validate.models.data_deserialised import ResourceData
+from dsp_tools.commands.xml_validate.models.data_deserialised import RichtextData
+from dsp_tools.commands.xml_validate.models.data_deserialised import SimpleTextData
+from dsp_tools.commands.xml_validate.models.data_deserialised import TimeValueData
+from dsp_tools.commands.xml_validate.models.data_deserialised import UriValueData
+from dsp_tools.commands.xml_validate.models.data_deserialised import ValueData
+
+
+def transform_into_project_deserialised(root: etree._Element) -> DataDeserialised:
+    """
+    Takes the root of an XML
+    Extracts the metadata of the project and transforms all its resources.
+
+    Args:
+        root: root of an xml with qnames and comments removed
+
+    Returns:
+        Class instance with the information reformatted
+    """
+    shortcode = root.attrib["shortcode"]
+    default_ontology = root.attrib["default-ontology"]
+    resources = _deserialise_all_resources(root)
+    return DataDeserialised(shortcode=shortcode, default_onto=default_ontology, resources=resources)
+
+
+def _deserialise_all_resources(root: etree._Element) -> list[ResourceData]:
+    all_res = []
+    for res in root.iterchildren():
+        match res.tag:
+            case "resource":
+                all_res.append(_deserialise_one_resource(res))
+            case "annotation" | "region" | "link" | "video-segment" | "audio-segment":
+                all_res.append(_deserialise_dsp_base_resource(res))
+            case _:
+                pass
+    return all_res
+
+
+def _deserialise_dsp_base_resource(resource: etree._Element) -> ResourceData:
```

Comment:
The validation of the base resources is part of milestone 3. But it is possible that other resources link to them, so they must exist in the data graph.